front/components/members/InvitationModal.tsx

```diff
@@ -0,0 +1,521 @@
+            <div className="font-semibold">
+              Email addresses (comma or newline separated):
+            </div>
+            <div className="flex items-start gap-2">
+              <div className="flex-grow">
+                <TextArea
+                  placeholder="Email addresses, comma or newline separated"
+                  value={inviteEmails}
+                  onChange={(value) => {
+                    setInviteEmails(value);
+                    setEmailError("");
+                  }}
+                  error={emailError}
+                  showErrorLabel={true}
+                />
+              </div>
+            </div>
+            <div className="flex flex-col gap-2">
+              <div className="flex items-center gap-2">
+                <div className="font-semibold text-element-900">Role:</div>
+                <RoleDropDown
+                  selectedRole={invitationRole}
+                  onChange={setInvitationRole}
+                />
+              </div>
+            </div>
+            <div className="text-element-700">
+              {ROLES_DATA[invitationRole]["description"]}
+            </div>
+          </div>
+          {isProPlanCode(plan.code) && (
+            <div className="justify-self-end">
+              <ProPlanBillingNotice />
+            </div>
+          )}
+        </div>
+      </Modal>
+    </>
+  );
+}
+
+export function EditInvitationModal({
+  owner,
+  invitation,
+  isOpen,
+  setOpen,
+}: {
+  owner: WorkspaceType;
+  invitation: MembershipInvitationType;
+  isOpen: boolean;
+  setOpen: (open: boolean) => void;
+}) {
+  const [invitationRole, setInvitationRole] = useState<ActiveRoleType>(
+    invitation.initialRole
+  );
+  const sendNotification = useContext(SendNotificationsContext);
+  const confirm = useContext(ConfirmContext);
+
+  function closeModal() {
+    setOpen(false);
+  }
+
+  return (
+    <Modal
+      title="Edit Invitation"
+      isOpen={isOpen}
+      onClose={closeModal}
+      saveLabel="Save"
+      savingLabel="Saving..."
+      hasChanged={false}
+      variant="side-sm"
+    >
+      <Page variant="modal">
+        <Page.Layout direction="vertical">
+          <Page.Layout direction="horizontal" sizing="grow" gap="sm">
+            <Page.H variant="h6">{invitation.inviteEmail}</Page.H>
+          </Page.Layout>
+          <div className="grow font-normal text-element-700">
+            Invitation sent on{" "}
+            {new Date(invitation.createdAt).toLocaleDateString()}
+          </div>
+          <div className="flex items-center gap-2">
+            <div className="font-semibold text-element-900">Role:</div>
+            <RoleDropDown
+              selectedRole={invitationRole}
+              onChange={async (invitationRole) => {
+                await handleInvitationRoleChange(
+                  owner,
+                  invitation,
+                  invitationRole
+                );
+                setInvitationRole(invitationRole);
```

Comment:
Is @Duncid ok with a dropdown that waits on the network? Do we have similar previous art? Do we weant the `setInvitationRole` before or after the network call?

Reply:
In fact you don't want the invitationRole state since the state comes from `invitation` and will get updated as per your implementation of `handleInvitationRoleChange` (which mutates at the end)

Reply:
Good point